packages/components/autocomplete/src/use-autocomplete.ts

```diff
@@ -278,6 +280,13 @@ export function useAutocomplete<T extends object>(originalProps: UseAutocomplete
         color: isInvalid ? "danger" : originalProps?.color,
         isIconOnly: true,
         disableAnimation,
+        onClick: () => {
+          // if the listbox is open, clicking selector button should
+          // close the listbox and focus on the input
+          if (state.isOpen) {
+            setShouldFocus(true);
+          }
+        },
       },
```

Comment:
@wingkwong merge this onClick with the `selectorButtonProps.onClick`, 

e.g

`onClick: chain(currentOnClickFunction, selectorButtonProps.onClick)